sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -319,7 +320,8 @@ abstract class StreamExecution(
         // This is a workaround for HADOOP-12074: `Shell.runCommand` converts `InterruptedException`
         // to `new IOException(ie.toString())` before Hadoop 2.8.
         updateStatusMessage("Stopped")
-      case e: Throwable =>
+      case t: Throwable =>
+        val e = QueryExecution.toInternalError(msg = s"Execution of the stream $name failed.", t)
```

Comment:
do you know why v2 streaming sink doesn't need this?

Reply:
v2 invokes `executedPlan()` at:
https://github.com/apache/spark/blob/8fab5978e30db4f2420d7830956462bd3e50ef33/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala#L651
in which we convert the exception.

but v1 gets the exception at another line:
https://github.com/apache/spark/blob/8fab5978e30db4f2420d7830956462bd3e50ef33/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala#L548